### ISSUE TITLE
Failed to build dropout-layer-norm

### ISSUE THREAD

**Issue by vgoklani** [Dao-AILab/flash-attention#587]
Hey there,

I'm not able to build the dropout-layer-norm.

I'm using the Docker image: nvcr.io/nvidia/pytorch:23.09-py3

and then just installing the flash attention components via:

    flash_attn_version=2.3.0

    pip install flash-attn==${flash_attn_version}

    cd

    git clone https://github.com/HazyResearch/flash-attention \
        && cd flash-attention && git checkout v${flash_attn_version} \
        && cd csrc/fused_softmax && pip install . && cd ../../ \
        && cd csrc/rotary && pip install . && cd ../../ \
        && cd csrc/xentropy && pip install . && cd ../../ \
        && cd csrc/fused_dense_lib && pip install . && cd ../../ \
        && cd csrc/ft_attention && pip install . && cd ../../ \
        && cd .. && rm -rf flash-attention

these are the device specs:

    gpu:0 - NVIDIA RTX 6000 Ada Generation
    device_properties.name NVIDIA RTX 6000 Ada Generation
    multi-processor-count: 142
    gpu:1 - NVIDIA RTX 6000 Ada Generation
    device_properties.name NVIDIA RTX 6000 Ada Generation
    multi-processor-count: 142
    PyTorch version：2.1.0a0+32f93b1
    CUDA Version: 12.2
    cuDNN version is: 8905
    Arch version is: sm_52 sm_60 sm_61 sm_70 sm_72 sm_75 sm_80 sm_86 sm_87 sm_90 compute_90
    ARCH LIST: ['sm_52', 'sm_60', 'sm_61', 'sm_70', 'sm_72', 'sm_75', 'sm_80', 'sm_86', 'sm_87', 'sm_90', 'compute_90']
    Device Capability: (8, 9)

thanks!

**Comment by vgoklani** [comment]
<bump>

**Comment by Lvjinhong** [comment]
I've tried installing flash-attn using pip install flash-attn==2.2.1 and flash-attn==2.3. However, when I attempt distributed training with Megatron LM, I consistently encounter the following issue:

![image](https://github.com/Dao-AILab/flash-attention/assets/96970081/5ab8b9be-948f-4c58-8a77-0eb4638bd49f)

Additionally, when I tried building from the source code, the issue persisted.

**Comment by tridao** [comment]
dropout_layer_norm is a separate extension. You don't have to use it.

**Comment by vgoklani** [comment]
@tridao to be clear, we want to use it :) but it's not building correctly

from above

    && cd csrc/layer_norm && pip install . && cd ../../ \

**Comment by sipie800** [comment]
same here. failed in generating the obj files

**Comment by tridao** [comment]
As mentioned in https://github.com/Dao-AILab/flash-attention/tree/main/csrc/layer_norm:
As of 2024-01-05, this extension is no longer used in the FlashAttention repo. We've instead switched to a Triton-based [implementation](https://github.com/Dao-AILab/flash-attention/blob/main/flash_attn/ops/triton/layer_norm.py).

**Comment by sipie800** [comment]
> As mentioned in https://github.com/Dao-AILab/flash-attention/tree/main/csrc/layer_norm: As of 2024-01-05, this extension is no longer used in the FlashAttention repo. We've instead switched to a Triton-based [implementation](https://github.com/Dao-AILab/flash-attention/blob/main/flash_attn/ops/triton/layer_norm.py).

thanks for replying. Does that mean that a model developer has to  modify their usage in a model of flash_attn to use triton one? Or flash_attn will switch it internally by itself?

**Comment by tridao** [comment]
Internally we already use the Triton implementation for layernorm.

**Comment by sipie800** [comment]
I'm using Qwen LLM in modelscope frame. Have flash_attn 2.5.6 installed.

----
Try importing flash-attention for faster inference...
Warning: import flash_attn rms_norm fail, please install FlashAttention layer_norm to get higher efficiency https://github.com/Dao-AILab/flash-attention/tree/main/csrc/layer_norm
----

so might I just ignore the warnings above when modelscope is loading model?

**Comment by tridao** [comment]
Sorry I can't control what Qwen implementation uses.

**Comment by sipie800** [comment]
> Sorry I can't control what Qwen implementation uses.

That's true. Yet if flash_attn use triton layernorm internally, there should not be such a warnings? They are just calling layernorm, whether it's the triton one or old one? Or the old layernorm and triton one are not replaced actually.

**Comment by tridao** [comment]
The warning is printed from Qwen's code. I can't control that.

**Comment by qyr0403** [comment]
have you solve this problem？i also meet this

**Comment by htchentyut** [comment]
https://github.com/OpenGVLab/InternVideo/blob/main/InternVideo2/multi_modality/INSTALL.md
